src/Native/Windows/gen-buildsys-win.bat

```diff
@@ -8,18 +8,29 @@ for %%x in (%*) do Set /A argC+=1
 if NOT %argC%==3 GOTO :USAGE
 if %1=="/?" GOTO :USAGE
 
-setlocal
+setlocal EnableDelayedExpansion
 set __sourceDir=%~dp0
 set __ExtraCmakeParams=
 
-:: VS 2017 is required to build native assets,
-:: once CMake supports 2019 as the project generator we can use that if installed in the local system
-set __VSString=15 2017
+:: VS 2017 is the minimum supported toolset
+if "%__VSVersion%" == "vs2019" (
+  set __VSString=16 2019
 
-:: Set the target architecture to a format cmake understands. ANYCPU defaults to x64
-if /i "%3" == "x86"     (set __VSString=%__VSString%)
-if /i "%3" == "x64"     (set __VSString=%__VSString% Win64)
-if /i "%3" == "arm"     (set __VSString=%__VSString% ARM)
+  :: VS 2019 generator is not per arch, and the arch needs to be passed with -A option
+  :: x86 is the default arch selected by cmake.
+  :: CMAKE 3.14 or later is required to use VS2019
+  if /i "%3" == "x64"     (set __ExtraCmakeParams=%__ExtraCmakeParams% -A x64)
```

Comment:
Why not just use `-A` everywhere? IIRC, it should be supported on older CMake versions as well.

Reply:
But arch specific generators might do special stuff depending on the arch?

Reply:
No, the `-A` option and the legacy `target platform option` are the same: https://cmake.org/cmake/help/v3.13/generator/Visual%20Studio%2015%202017.html

Reply:
Nice! Great find. I will change it to use -A always instead.